### PR TITLE
pkg: u8g2: improvements and bugfixes

### PR DESCRIPTION
**Pull request by basilfx**
Just a few bug fixes and improvements due to [new developments](https://github.com/olikraus/u8g2/pull/37) that should simplify a few things. I also simplified the test application.

Mark it as WIP, because I still have to test one display, which I don't have acces to. In the mean time, maybe @aabadie wants to check out the improvements?

### CODE REVIEW

**Comment by aabadie** [comment]
@basilfx, looking briefly at the changes, they look good to me. I'll give it a try when time permits.

**Comment by basilfx** [comment]
The display I wanted to test (LS013B7DH03) works, so it's not WIP anymore :-)

File: tests/pkg_u8g2/Makefile [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
+TEST_ADDR ?= 0x78
 
-TEST_U8G2_I2C       ?= 0
-TEST_U8G2_SPI       ?= 0
+TEST_PIN_CS ?= GPIO_PIN\(0,0\)
+TEST_PIN_DC ?= GPIO_PIN\(0,0\)
+TEST_PIN_RESET ?= GPIO_PIN\(0,0\)
 
-TEST_U8G2_ADDR      ?= 0x78
+ifeq ($(TEST_OUTPUT),3)
```
Comment (aabadie) [inline tests/pkg_u8g2/Makefile]: what do you think of merging this case with the same one a few lines below ? same for the I2C case.

Comment (basilfx) [inline tests/pkg_u8g2/Makefile]: I can move that one inside, as it is only required for I2C.

To make it clear what options you need per output type, I have divided it into multiple if-statements on purpose.

**Comment by basilfx** [comment]
Applied fix from #5737 and squashed into two logical commits.

**Comment by aabadie** [comment]
@basilfx, sorry I forgot this one. Will give it a shot asap.

**Comment by aabadie** [comment]
Just tested it on my I2C OLED module.
As a reminder I built and flashed the `pkg_u8g2` on a samr21-xpro board with the following command:
```
make BOARD=samr21-xpro TEST_DISPLAY=u8g2_Setup_ssd1306_i2c_128x64_noname_1 TEST_OUTPUT=4 TEST_I2C=0 TEST_ADDR=0x3C clean flash term
```
It still works.

The default display on the terminal also still works.

=> ACK, ready for CI build

**Comment by aabadie** [comment]
@kaspar030, it seems Murdock is stuck from a while now, is there a problem ?
@basilfx can you amend the last commit and force push to trigger a new run on Murdock ?

**Comment by aabadie** [comment]
Murdock is Happy. Ack and GO !

**Pull request merged**